tests/test-opt-out-integration.py

```diff
@@ -148,11 +149,20 @@ def test_UploadEvents_fails_if_disabled(self):
             self.interface.UploadEvents()
 
     def _check_config_file(self, enabled, uploading_enabled):
+        # the config file is written asyncronously by the daemon,
+        # so may not exist at the start of a test
+        while not os.path.exists(self.config_file):
```

Comment:
Probably not a good idea to have a potentially unbounded while loop, as it will get stuck here forever if an error prevents the path from ever existing.